rotkehlchen/chain/ethereum/modules/liquity/trove.py

```diff
@@ -0,0 +1,506 @@
+                # In one transaction is possible to generate debt and change the colateral
+                if debt_change != AssetAmount(ZERO):
+                    if debt_change > ZERO:
+                        # Generate debt
+                        count_spent_got_cost_basis = True
+                        got_asset = A_LUSD
+                        got_balance = Balance(
+                            amount=debt_change,
+                            usd_value=query_usd_price_or_use_default(
+                                A_LUSD,
+                                timestamp,
+                                ZERO,
+                                'Liquity',
+                            ),
+                        )
+                        total_lusd_trove_balance += got_balance
+                    else:
+                        # payback debt
+                        count_spent_got_cost_basis = True
+                        spent_asset = A_LUSD
+                        spent_balance = Balance(
+                            amount=abs(debt_change),
+                            usd_value=query_usd_price_or_use_default(
+                                A_LUSD,
+                                timestamp,
+                                ZERO,
+                                'Liquity',
+                            ),
+                        )
+                        total_lusd_trove_balance -= spent_balance
+                        balance = total_lusd_trove_balance.amount + realized_trove_lusd_loss.amount
+                        if balance < ZERO:
+                            pnl_balance = total_lusd_trove_balance + realized_trove_lusd_loss
+                            realized_trove_lusd_loss += -pnl_balance
+                            pnl = [AssetBalance(asset=A_LUSD, balance=pnl_balance)]
+                if collateral_change != AssetAmount(ZERO):
+                    if collateral_change < ZERO:
+                        # Withdraw collateral
+                        got_asset = A_ETH
+                        got_balance = Balance(
+                            amount=abs(collateral_change),
+                            usd_value=query_usd_price_or_use_default(
+                                A_ETH,
+                                timestamp,
+                                ZERO,
+                                'Liquity',
+                            ),
+                        )
+                    else:
+                        # Deposit collateral
+                        spent_asset = A_ETH
+                        spent_balance = Balance(
+                            amount=collateral_change,
+                            usd_value=query_usd_price_or_use_default(
+                                A_ETH,
+                                timestamp,
+                                ZERO,
+                                'Liquity',
+                            ),
+                        )
+                if operation in (
```

Comment:
All this processing may or may not be correct. Looks okay from first view but would need to test with real data

Reply:
Separate to make it more readable

```suggestion

                if operation in (
```